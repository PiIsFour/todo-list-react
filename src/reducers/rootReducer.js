import { combineReducers } from './util'
import todoListsReducer from './todoListsReducer'

const rootReducer = combineReducers({
	todoLists: todoListsReducer,
})

export default rootReducer
