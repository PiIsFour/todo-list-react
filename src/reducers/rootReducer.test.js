import rootReducer from './rootReducer'

describe('rootReducer', () => {
	it('initial state matches snapshot', () => {
		const initialState = rootReducer()
		expect(initialState).toMatchSnapshot()
	})

	it('returns the same state if none of the substates change', () => {
		const initialState = rootReducer()
		const unusedAction = { type: 'SHOULD_NOT_BE_USED' }
		expect(rootReducer(initialState, unusedAction)).toBe(initialState)
	})

	it('to contain an empty todolists as default', () => {
		const initialState = rootReducer()
		expect(initialState).toEqual(expect.objectContaining({
			todoLists: [],
		}))
	})
})
