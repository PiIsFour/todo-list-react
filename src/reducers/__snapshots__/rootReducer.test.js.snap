// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootReducer initial state matches snapshot 1`] = `
Object {
  "todoLists": Array [],
}
`;
